### ISSUE TITLE
Multi-speaker TTS

### ISSUE THREAD

**Issue by arunraja-hub** [TensorSpeech/TensorFlowTTS#466]
In [Towards End-to-End Prosody Transfer for Expressive Speech Synthesis with Tacotron](https://arxiv.org/pdf/1803.09047.pdf), the authors have implemented multi-speaker models by conditioning on speaker identity.

However, in [E2E-TensorflowTTS Demo colab notebook](https://colab.research.google.com/drive/1akxtrLZHKuMiQup00tzO2olCaN-y3KiD?usp=sharing), the example only gives the audio output for one speaker. 

How can I generate multi-speaker audio outputs using TensorflowTTS? Is there any colab notebook that illustrates this? Any code and implementation advice would be useful.

**Comment by ZDisket** [comment]
@arunraja-hub The mel generation models can do multispeaker, you just have to modify the dataloader. You can see how I did it for FastSpeech2 [here](https://github.com/ZDisket/TensorflowTTS/commit/616a70cb3981771bdfcd44e8a1d7c5f5dd7d1fd5#diff-9a53aa155125bb4b5f51f8d85f2fe438ca461b7ec88dfa27acd04dfd8712e790). This one takes a `-speakers.npy` file for every utt id containing `[speaker_id]` as numpy int32 in a dedicated folder in each dump (val, train). You also have to modify the config for training and use the appropriate speaker id at inference time. You'll have to write the script that outputs those files yourself.
There's also a multispeaker LibriTTS example here: https://github.com/TensorSpeech/TensorFlowTTS/tree/master/examples/fastspeech2_libritts

**Comment by arunraja-hub** [comment]
Thank you very much! 
Another question: Is it possible to produce multi-speaker data by ourselves or is the compute power required only feasible by Google?
I just want to get my expectations right regarding producing multi-speaker data for the [MGB-1 dataset](http://www.mgb-challenge.org/MGB-1.html)

**Comment by ZDisket** [comment]
@arunraja-hub By producing data I assume you mean training model. And no, training doesn't require that much computing power, one V100 which you can rent for cheap hourly rates from various sites is enough.

**Comment by yolanda0235** [comment]
@ZDisket  @dathudeptrai 
I modified the processor , mixed ljspeech dataset and Baker dataset as English and Chinese data, and assigned them speaker ID, （Chinese is 0, English is 1）

![Selection_001](https://user-images.githubusercontent.com/50702348/108477319-07863f00-72ce-11eb-818a-daddd8a92cff.png)

I modified  tacotron2 model according to the example of multispeaker libritts. Speakerid is added to the training process.
![Selection_002](https://user-images.githubusercontent.com/50702348/108477509-44eacc80-72ce-11eb-9dda-b05ea1c08ee1.png)


But when I give the speaker ID [0] or [1] in inference, I only get the voice of speaker ID 0, which is totally incorrect in English

What did I do wrong? Any suggestion?
Any help to resolve this would be greatly appreciated!
